### DISCUSSION TITLE
@ export "there is no route"

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#107] [Q&A] [answered]
It frequently happens that when a file is to be exported, the message "there is no route" appears.  The file export is empty.

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the report!

Is the video with the latest version **3.0.15**?

**Reply by 0709wiwiwi** [reply]
3.0.15

**Comment by devemux86** [comment]
I could not reproduce it or have seen it before, that is strange.

Anyway the import / export was rewritten to introduce save of their options,
so let's see how it is in the next Cruiser version.

- https://github.com/devemux86/cruiser/discussions/49

**Comment by 0709wiwiwi** [comment]


**Reply by devemux86** [reply]
@0709wiwiwi thanks for the extra details!

The failed export (with the custom name) seems to cause the app screen to flash, like the app restarts?
There is no flash in the app screen after the successful export.

What happens with a custom name _without spaces_?

**Comment by devemux86** [comment]
@0709wiwiwi please check with [Cruiser 3.0.16](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

The export route was rewritten, it should work more reliably now.

**Reply by 0709wiwiwi** [reply]
The problem still occurs.

**Reply by devemux86** [reply]
@0709wiwiwi thanks for the tests!

What folder path do you choose to export?

In the Android file browser that opens and in its left drawer,
the default folder shortcuts (Downloads, Recent, ...) often have problems and show deleted files!

It is better to click on the root of the internal or external storage and browse to the preferred folder.
Use the green folders (not the red shortcut):

<img src="https://user-images.githubusercontent.com/3484020/230576175-b1bcf579-cfb2-4fa2-834d-08ca6951ccfd.jpg" width="50%"/>

**Comment by 0709wiwiwi** [comment]


**Reply by devemux86** [reply]
@0709wiwiwi 

The #340 was related with the changes for the new [favorite tracks](https://github.com/devemux86/cruiser/discussions/81).

Have you checked what I mentioned above in https://github.com/devemux86/cruiser/discussions/107#discussioncomment-5551650?
Does it happen with the menu "Routing | Export"?
What export format do you select?

**Comment by 0709wiwiwi** [comment]
**Does it happen with the menu "Routing | Export"?** Yes
**What export format do you select?** Gpx.

**Reply by devemux86** [reply]
@0709wiwiwi where do you export, internal storage or sd card?

And how do you select the export location?
(please see my comment above in https://github.com/devemux86/cruiser/discussions/107#discussioncomment-5551650)

**Comment by devemux86** [comment]
@0709wiwiwi 

Checking the video again, it looks like that phone when it returns from the file selection, starts Cruiser again?!
But Cruiser is already running in the background, no need to start it again. So the active route is not ready yet.

Probably another Android change by OEM...?

I will adapt slightly the export (internal) process and you can test it again with the next Cruiser update.

**Reply by DolomitesTom** [reply]
By the way is it still not solveable that the export name is always date and time and not the name of the before openened file?

**Reply by devemux86** [reply]
@DolomitesTom please see https://github.com/devemux86/cruiser/discussions/348.

**Reply by devemux86** [reply]
@0709wiwiwi can you test on your devices the latest:
- https://github.com/devemux86/cruiser/discussions/329

**Reply by 0709wiwiwi** [reply]
All exports are fine. (No error)

**Reply by devemux86** [reply]
@0709wiwiwi thanks for the tests and the report! :+1:

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
